src/System.Private.CoreLib/shared/System/DateTimeOffset.cs

```diff
@@ -553,8 +553,8 @@ void IDeserializationCallback.OnDeserialization(object sender)
         {
             try
             {
-                _offsetMinutes = ValidateOffset(Offset);
-                _dateTime = ValidateDate(ClockDateTime, Offset);
+                ValidateOffset(Offset);
+                ValidateDate(ClockDateTime, Offset);
```

Comment:
```csharp
Offset => new TimeSpan(0, _offsetMinutes, 0);
ClockDateTime => new DateTime((_dateTime + Offset).Ticks, DateTimeKind.Unspecified)

long utcTicks = ClockDateTime.Ticks - Offset.Ticks;
return  new DateTime(utcTicks, DateTimeKind.Unspecified);
```
Again if the validation passes, it will just return the original value; so the assignment is redundant (also prevents it being readonly)